src/Provider/IslandoraServiceProvider.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Islandora\Crayfish\Commons\Provider;
+
+use Islandora\Crayfish\Commons\CmdExecuteService;
+use Islandora\Crayfish\Commons\FedoraResourceConverter;
+use Pimple\Container;
+use Pimple\ServiceProviderInterface;
+use Silex\Provider\DoctrineServiceProvider;
+use Silex\Provider\MonologServiceProvider;
+use Silex\Provider\ServiceControllerServiceProvider;
+use Silex\Provider\SecurityServiceProvider;
+use Islandora\Chullo\FedoraApi;
+use Islandora\Crayfish\Commons\Syn\SettingsParser;
+use Islandora\Crayfish\Commons\Syn\JwtAuthenticator;
+use Islandora\Crayfish\Commons\Syn\JwtFactory;
+
+class IslandoraServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function register(Container $container)
+    {
+        // Register services we rely on
+        $container->register(new MonologServiceProvider());
+        $container->register(new ServiceControllerServiceProvider());
+        $container->register(new SecurityServiceProvider());
+        $container->register(new DoctrineServiceProvider());
+
+        // Configure external services
+        $container['monolog.logfile'] = function ($container) {
+            return strtolower($container['crayfish.log.level']) == 'none' ? null : $container['crayfish.log.file'];
+        };
+        $container['monolog.level'] = function ($container) {
+            return $container['crayfish.log.level'];
```

Comment:
So `YamlConfigServiceProvider` puts all the configuration from a particular YAML file into a "namespace". The [default namespace](https://github.com/Islandora-CLAW/Crayfish-Commons/pull/8/files#diff-4724351e1c9a88f9009ec48cc7a65aadR15) is crayfish. It then loads every key in the YAML file [concatenating them together](https://github.com/Islandora-CLAW/Crayfish-Commons/pull/8/files#diff-4724351e1c9a88f9009ec48cc7a65aadR23) with dots. 

So for example: 
```yaml
---
test:
  for: jared
```

Would make a new entry in the service container: `crayfish.entry.for`. This would have a value of `jared`. That is why all the configuration is loading from the crayfish namespace. 

So for the issue you are having here is that the `crayfish.log.level` variable isn't defined on the container. It should be defined [here](https://github.com/Islandora-CLAW/Crayfish/pull/27/files#diff-5b9d10f1edddae11ff8a90bb322931d4R15). Do you have it defined in your config file?